### PR TITLE
GitHub Actions refactor

### PR DESCRIPTION
**Pull request by legleux**
Run `clang-format`

Adds builds for:
- GCC 11
- Clang 14
- AppleClang 15

Runs unit tests
Removed Docker image publish. Adding it in a new action in subsequent PR.
Moved Linux packaging to different repository

### CODE REVIEW

**Comment by legleux** [comment]
[macOS failure](https://github.com/ripple/xbridge_witness/actions/runs/9083346308/job/24961842278?pr=105) is expected at the moment.

Hopefully this addition can prevent that in the future.

File: src/xbwd/app/main.cpp [hunk 1]
```diff
@@ -127,7 +127,8 @@ main(int argc, char** argv)
 
     try
     {
-        std::unique_ptr<xbwd::config::Config> config = [&]() -> auto {
+        std::unique_ptr<xbwd::config::Config> config = [&]() -> auto
```
Comment (oleks-rip) [inline src/xbwd/app/main.cpp:130]: Original version is correctly clang-formatted.

File: src/xbwd/app/main.cpp [hunk 2]
```diff
@@ -143,7 +144,8 @@ main(int argc, char** argv)
             if (!Json::Reader().parse(f, jv))
                 throw std::runtime_error("config file contains invalid json");
             return std::make_unique<xbwd::config::Config>(jv);
-        }();
+        }
+        ();
```
Comment (oleks-rip) [inline src/xbwd/app/main.cpp:148]: Same here. It was already clang-formatted. Change is not clang formatted.

**Review by oleks-rip — CHANGES_REQUESTED** [review]

File: src/xbwd/app/main.cpp (hunk 2 above)
Comment (legleux) [reply src/xbwd/app/main.cpp:148]: This _does_ look like a peculiar formatting decision.
Is the config not being used or is the command being run not correct?
I just copied what `rippled` has. This is `clang-format-10`

```bash
find src/xbwd -type f \( -name '*.cpp' -o -name '*.h' -o -name '*.ipp' \) -print0 | xargs -0 clang-format-${CLANG_VERSION} -i
```
https://github.com/legleux/xbridge_witness/blob/4f9643a4fae1b9cbb3c0c26244fb231f7bf0655d/.github/workflows/clang-format.yml#L23

File: src/xbwd/app/main.cpp (hunk 1 above)
Comment (legleux) [reply src/xbwd/app/main.cpp]: Looks like that patch fixed up the weird dangling `();`

**Review by oleks-rip — APPROVED** [review]

**Review by seelabs — APPROVED** [review]

**Pull request merged**